### DISCUSSION TITLE
jsPsych.currentTrial and event handling

### DISCUSSION THREAD

**Discussion by rouderj** [jspsych/jsPsych#2422] [Q&A]
I am trying to adapt the psychophysics event handling demos for gabor patch parameters.  The demos transfer seamlessly to some gabor parameters (i.e. startX), but not to others (i.e., tilt).  I am trying to learn javascript, jsPsych, and plugins at the same time, so if this is a dumb question, I apologize.  I suspect the problem, however, is not my stupidity (this time), but that only some parameters can be updated through jsPsych.currentTrial and not others.  I can read the correct parameter value in the console log, so I dont think it is an indexing issue.  Thx....

Here is an example that changes X position but not tilt:

```js
   let current = 90;

      var circle_obj = {
        obj_type: 'gabor',
        startX: current,
        startY: 150,
        tilt: current,
        sf: 0.05,
        phase: 90,
        width: 150,
        sc: 10,
        contrast: 0.5,
        disableNorm: true
      };

      const trial = {
          type: 'psychophysics',
          canvas_height: 500,
          prompt: '<p>Pressing the ArrowUp/ArrowDown key, the color of the circle will change. <br>Press the space key to finish the program.</p>',
          stimuli: [circle_obj], // These can be referenced using the jsPsych.currentTrial().stimuli array.
          response_type: 'key',
          choices: [' '],
          key_down_func: function(event){ // The key_up_func is also available. In that case, the color of the circle changes when you release the key. 
            if (event.key === 'ArrowUp'){
              current += 2;
            } else if (event.key === 'ArrowDown'){
              current -= 2;
            }

            // Note that when specify the space bar, you need to write event.key === ' ' not event.key === 'space'.
            
            jsPsych.currentTrial().stim_array[0].startX = current;
            jsPsych.currentTrial().stim_array[0].tilt = current;
          },
      }
```

**Comment by jodeleeuw** [comment]
Going to tag @kurokida in case he isn't subscribed to discussions notifications.

**Reply by rouderj** [reply]
Thx!

**Reply by kurokida** [reply]
Hi @jodeleeuw ,thank you for tagging me.
I answered on https://github.com/kurokida/jspsych-psychophysics/issues/27#issuecomment-1004478650.
